cmd/kubeadm/app/util/staticpod/utils.go

```diff
@@ -228,6 +230,19 @@ func ReadinessProbe(host, path string, port int, scheme v1.URIScheme) *v1.Probe
 	return createHTTPProbe(host, path, port, scheme, 0, 15, 3, 1)
 }
 
+// StartupProbe creates a Probe object with a HTTPGet handler
+func StartupProbe(host, path string, port int, scheme v1.URIScheme, timeoutForControlPlane *metav1.Duration) *v1.Probe {
+	periodSeconds := int32(10)
+	if timeoutForControlPlane == nil {
```

Comment:
Let's do:
```golang
totalDuration := constants.DefaultControlPlaneTimeout.Seconds()
if timeoutForControlPlane != nil {
	totalDuration = timeoutForControlPlane.Seconds()
}
```

to avoid the memory allocation.

Reply:
Sorry for the (maybe) dumb question but shouldn't `timeoutForControlPlane` which is in fact `cfg.APIServer.TimeoutForControlPlane` be either the value that the user has set or the default value? I mean in this function the default value is set: https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/v1beta1/defaults.go#L107-L113 so do we really need this `nil` check?

Reply:
For some test cases in https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/phases/controlplane/manifests_test.go, we create the `ClusterConfiguration` without `APIServer` field(such as https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/phases/controlplane/manifests_test.go#L49-L51) as we do not care about what it is.
If we do not do the `nil` check, these tests will fail. We should always guarantee the function work well no matter what is inputted.

Reply:
That makes sense, thanks for your clarification :)